src/service-worker.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import {writeFile} from 'fs';
+import * as path from 'path';
+import * as logging from 'plylog';
+import {PolymerProject} from './polymer-project';
+import {DepsIndex} from './analyzer';
+const swPrecache = require('sw-precache');
+
+// non-ES compatible modules
+let logger = logging.getLogger('polymer-build.service-worker');
+
+export interface SWConfig {
```

Comment:
Is this interface just describing an interface of the sw-precache library? If so, it should live in custom_typings, and they you can import sw-precache as a module